src/apps/chifra/pkg/file/lock.go

```diff
@@ -0,0 +1,41 @@
+package file
+
+import (
+	"io"
+	"os"
+	"syscall"
+)
+
+func Lock(file *os.File) error {
+	// Lock configuration. We want both read (F_RDLCK) and
+	// write (F_WRLCK) locks, and by Start to be interpreted
+	// as the start of the file (Whence set to SEEK_SET).
+	//
+	// More: https://man7.org/linux/man-pages/man2/fcntl.2.html
+	lockConfig := &syscall.Flock_t{
+		Type:   syscall.F_RDLCK | syscall.F_WRLCK,
+		Whence: int16(io.SeekStart),
```

Comment:
According to fcntl (2) manual:
```
short l_whence;  /* How to interpret l_start:
                        SEEK_SET, SEEK_CUR, SEEK_END */
```
So I gather that `whence` is not related to the size of a file, but it's rather just a flag. It's int16 because that's what golang wants...